src/org/jitsi/impl/neomedia/rtp/RTPEncodingDesc.java

```diff
@@ -280,55 +280,84 @@ else if (tsDiff >= (-(1L << 30) & 0xFFFFFFFFL))
             int bMinSeqNum = b.getMinSeen(), aMaxSeqNum = a.getMaxSeen();
             int snDiff = (bMinSeqNum - aMaxSeqNum) & 0xFFFF;
 
-            if (bFirstSeqNum != -1 || aLastSeqNum != -1)
+            boolean guessed = false;
+
+            // We have some codecs (VPX) for which we can inspect the payload
+            // to deduce the start/end of a frame. This has an impact on the
+            // distances between the frames that allow us to guess frame
+            // boundaries.
+            boolean withFrameBoundaries =
+                a.supportsFrameBoundaries() && b.supportsFrameBoundaries();
+
+            if (withFrameBoundaries)
             {
-                if (snDiff == 2)
+                if (bFirstSeqNum != -1 || aLastSeqNum != -1)
                 {
-                    if (aLastSeqNum == -1)
+                    if (snDiff == 2 || snDiff == (-2 & 0xFFFF))
```

Comment:
I like your idea about including the picture ID in our heuristics, but.. we don't have it in FrameDesc and I feel it doesn't belong there. I feel we may need to subclass it into VPXFrameDesc and H264FrameDesc and move the heuristics logic in there.

I added a note about your idea, but I'd rather not implement it now.